blockchain/pallets/minting/src/tests.rs

```diff
@@ -282,6 +287,63 @@ mod register_identity {
         });
     }
 
+    #[test]
+    fn first_call_to_register_for_minting_is_free() {
+        use frame_support::pallet_prelude::Pays;
+
+        new_test_ext().execute_with(|| {
+            register_id_account(42, 1);
+
+            // first call is free, actual_weight set to 0
+            let tree_0 = gen_tree(&["a", "b"]);
+            assert_eq!(
+                FractalMinting::register_for_minting(Origin::signed(1), Some(42), tree_0),
+                Ok(PostDispatchInfo {
+                    actual_weight: None,
+                    pays_fee: Pays::No
+                })
+            );
+
+            // second is charged, actual_weith None indicates annotation
+            // weight is applied
+            let tree_1 = gen_tree(&["a", "b", "c"]);
+            assert_eq!(
+                FractalMinting::register_for_minting(Origin::signed(1), Some(42), tree_1),
+                Ok(PostDispatchInfo {
+                    actual_weight: None,
+                    pays_fee: Pays::Yes
+                })
+            );
+        });
+    }
+
+    #[test]
+    fn first_call_to_register_for_minting_is_free_logic() {
```

Comment:
I don't know what this test is supposed to be testing from this name.

Reply:
I agree, this is now covered in second test I created from previous remark